### PR TITLE
Remove iOS 11 support.

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

PR removes iOS 11 support.

### CODE REVIEW

**Review by 1ec5 — COMMENTED** [review]
Also remove availability guards for iOS 12.0:

https://github.com/mapbox/mapbox-navigation-ios/blob/fb9dd5bd80bb6d34724d441640a06ab9e49e990f/Sources/MapboxNavigation/GenericRouteShield.swift#L100

Suggested changelog entry:

* This library now requires a minimum deployment target of iOS 12.0 or above. iOS 11._x_ is no longer supported. ([#4142](https://github.com/mapbox/mapbox-navigation-ios/pull/4142))

**Comment by 1ec5** [comment]
> please also change this text in docs.mapbox.com

That looks like the map SDK documentation. This PR only concerns the navigation SDK but doesn’t necessarily require us to increase the map SDK’s minimum deployment target.

**Comment by MaximAlien** [comment]
> 



> please also change this text in docs.mapbox.com
> 
> ```
> ## Requirements
> 
> The Mapbox Maps SDK for iOS is compatible with applications that are:
> 
> - Built in Xcode 13 or higher
> - Written in Swift version 5.3 or higher
> - Deployed on iOS 11 or higher
> ```
> 
> to
> 
> ```
> ## Requirements
> 
> The Mapbox Maps SDK for iOS is compatible with applications that are:
> 
> - Built in Xcode 13.4 or higher
> - Written in Swift version 5.6 or higher
> - Deployed on iOS 12 or higher
> ```

I'll address documentation changes in scope of different PR.

File: Cartfile.private [hunk 1]
```diff
@@ -1,3 +1,3 @@
 github "Quick/Quick" ~> 3.1.2
 github "Quick/Nimble" ~> 9.0.1
-github "pointfreeco/swift-snapshot-testing" ~> 1.8.0
+github "pointfreeco/swift-snapshot-testing" == 1.9.0
```
Comment (MaximAlien) [inline Cartfile.private:3]: Had to pin to 1.9.0 specifically, 1.10.0 no longer supports iOS 12.

**Pull request merged**